File: DFS+BFS/박성수/pro_타겟넘버.py
# https://school.programmers.co.kr/learn/courses/30/lessons/43165
answer = 0

def dfs(idx, numbers, target, value):
    global answer
    if (idx == len(numbers) and target == value):
        answer += 1
        return
    if (idx == len(numbers)):
        return
    dfs(idx + 1, numbers, target, value + numbers[idx])
    dfs(idx + 1, numbers, target, value - numbers[idx])


def solution(numbers, target):
    dfs(0, numbers, target, 0)
    return answer

File: DFS+BFS/전승훈/pro_타겟 넘버.swift
import Foundation

func solution(_ numbers:[Int], _ target:Int) -> Int {
    var ans = 0

    func DFS(idx: Int, sum: Int) {
        if idx == numbers.count {
            if sum == target {
                ans += 1
            }
            return
        }
        DFS(idx: idx+1, sum: sum + numbers[idx])
        DFS(idx: idx+1, sum: sum - numbers[idx])
    }

    DFS(idx: 0, sum: 0)

    return ans
}
